Criollo/Source/CRConnection.m

```diff
@@ -217,7 +220,11 @@ - (void)didFinishResponseForRequest:(CRRequest *)request {
     CRConnection * __weak connection = self;
     [self.delegate connection:self didFinishRequest:request response:request.response];
     dispatch_async(self.isolationQueue, ^{
-        [connection.requests removeObject:request];
+        if (connection != nil) {
+            @synchronized (connection) {
```

Comment:
This does not need a lock here, since it's being dispatched on the isolation queue. The purpose of this is to ensure synchronization without locking